torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -230,12 +234,17 @@ Tensor& resize_(
   }
   {
     at::AutoNonVariableTypeMode non_var_type_mode(true);
-    self_.resize_(size, std::move(optional_memory_format));
+    static auto op = c10::Dispatcher::singleton()
+      .findSchemaOrThrow("aten::resize_", "")
+      .typed<Tensor & (Tensor &, IntArrayRef, c10::optional<MemoryFormat>)>();
+    c10::Dispatcher::singleton()
+      .redispatch<Tensor &, Tensor &, IntArrayRef, c10::optional<MemoryFormat>>(op, ks & c10::DispatchKeySet(DispatchKeySet::FULL_AFTER, c10::DispatchKey::AutogradOther), self_, size, optional_memory_format);
```

Comment:
This is super wordy, we should figure out a way to more conveniently do redispatches without having to type all of this out.

Reply:
Some easy changes to make it less wordy/more readable that Basil pointed out are to make a constexpr for the updated DispatchKeySet and move the static op up top.

The ugliest part of this API though is having to manually input the function signature, and the <ReturnType, ParamTypes...>. One possibility that I took a quick turn at (and stopped due to compiler errors) was something like:
```
c10::redispatch<resize_>("aten::resize_", ks & c10::after_autograd_keyset, self_, size, optional_memory_format)
```

Where `redispatch` has a signature that's (something like)
```
Template<class typename FuncPtr, class Args...>
decltype(auto) redispatch(char* op_name, DispatchKeySet ks, Args...)
```
And we use `function_traits` to infer the parameter/returns of `FuncPtr` to appropriately pass in types to `\.typed<...>` and `.redispatch<...>.

If this still has issues and I can't think of a better approach, I could also potentially deal with this in a separate PR- since this API is almost entirely used in codegen for now (plus these three manual autograd kernels).

Reply:
I think at this point I'm OK with the three manually written sites in this PR for now. I agree that doing this is a project.

The function as template parameter approach is nice but unfortunately we would still need to generate a bunch of unambiguous function tokens for overloads, since the regular function names are ambiguous. But generally I am kind of positive on this sort of mechanism.